lib/email/sender.rb

```diff
@@ -388,9 +388,12 @@ def merge_json_x_header(name, value)
     end
 
     def set_reply_key(post_id, user_id)
-      return unless user_id &&
```

Comment:
Nitpick: I do not usually like defining variables only to be used immediately. I would have written a comment if I wanted to make the condition more descriptive.

I am also not a fan of `unless`, so what I would do is maybe change it with:

```ruby
return if !user_id ||
          !post_id ||
          header_value(Email::MessageBuilder::ALLOW_REPLY_BY_EMAIL_HEADER).blank?
```

Feel free to do go with any option, they all do the same. :)